encoding/protobuf.go

```diff
@@ -0,0 +1,98 @@
+package encoding
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/dynamic"
+	"github.com/yarpc/yab/transport"
+)
+
+// SplitMethod takes a method name like Service::Method and splits it
+// into Service and Method.
+func SplitMethod(fullMethod string) (svc, method string, err error) {
+	parts := strings.Split(fullMethod, "::")
+	switch len(parts) {
+	case 1:
+		return parts[0], "", nil
+	case 2:
+		return parts[0], parts[1], nil
+	default:
+		return "", "", fmt.Errorf("invalid proto method %q, expected form package.Service::Method", fullMethod)
+	}
+}
+
+type protoSerializer struct {
+	fullMethodName string
+	method         *desc.MethodDescriptor
+}
+
+// NewProtobuf returns a protobuf serializer.
+func NewProtobuf(fullMethodName string, source ProtoDescriptorSource) (Serializer, error) {
```

Comment:
This might be standard in this package, but if not, I'd prefer `NewProtobufSerializer`